src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/network/v4_9/CloudStackNetworkPlugin.java

```diff
@@ -80,73 +76,112 @@ public String requestInstance(NetworkOrder networkOrder, CloudStackUser cloudUse
+
+        try {
+            String jsonResponse = CloudStackCloudUtils.doRequest(
+                    this.client, uriRequest.toString(), cloudStackUser);
+
+            GetNetworkResponse response = GetNetworkResponse.fromJson(jsonResponse);
+            return getNetworkInstance(response);
+        } catch (HttpResponseException e) {
+            throw CloudStackHttpToFogbowExceptionMapper.get(e);
         }
     }
 
-    @Override
-    public void deleteInstance(NetworkOrder networkOrder, CloudStackUser cloudUser) throws FogbowException {
-        DeleteNetworkRequest request = new DeleteNetworkRequest.Builder()
-                .id(networkOrder.getInstanceId())
-                .build(this.cloudStackUrl);
+    @NotNull
+    @VisibleForTesting
+    String doRequestInstance(@NotNull CreateNetworkRequest createNetworkRequest,
+                             @NotNull CloudStackUser cloudStackUser)
+            throws FogbowException {
 
-        CloudStackUrlUtil.sign(request.getUriBuilder(), cloudUser.getToken());
+        URIBuilder uriRequest = createNetworkRequest.getUriBuilder();
+        CloudStackUrlUtil.sign(uriRequest, cloudStackUser.getToken());
 
         try {
-            this.client.doGetRequest(request.getUriBuilder().toString(), cloudUser);
+            String jsonResponse = CloudStackCloudUtils.doRequest(
+                    this.client, uriRequest.toString(), cloudStackUser);
+            CreateNetworkResponse response = CreateNetworkResponse.fromJson(jsonResponse);
+            return response.getId();
         } catch (HttpResponseException e) {
-            CloudStackHttpToFogbowExceptionMapper.map(e);
+            throw CloudStackHttpToFogbowExceptionMapper.get(e);
         }
     }
 
-    private String generateRandomNetworkName() {
-        return "fogbow_network_" + UUID.randomUUID();
-    }
-
-    private SubnetUtils.SubnetInfo getSubnetInfo(String cidrNotation) {
+    @NotNull
+    @VisibleForTesting
+    SubnetUtils.SubnetInfo getSubnetInfo(String cidrNotation) throws InvalidParameterException {
         try {
             return new SubnetUtils(cidrNotation).getInfo();
         } catch (IllegalArgumentException e) {
-            return null;
+            throw new InvalidParameterException(
+                    String.format(Messages.Exception.INVALID_CIDR, cidrNotation));
         }
     }
 
-    private NetworkInstance getNetworkInstance(GetNetworkResponse.Network network) {
+    @NotNull
+    @VisibleForTesting
+    NetworkInstance getNetworkInstance(@NotNull GetNetworkResponse response)
+            throws InstanceNotFoundException {
+
+        List<GetNetworkResponse.Network> networks = response.getNetworks();
+        if (networks.isEmpty()) {
+            throw new InstanceNotFoundException();
+        }
+        // since an id was specified, there should be no more than one network in the getNetworkResponse
+        GetNetworkResponse.Network network = networks.get(0);
```

Comment:
To avoid magic numbers like in the call "networks.get(0)" I suggest using "response.getNetworks().listIterator().next()" to get the first network of the list.

Reply:
Nice! I will fix this.